CMakeLists.txt

```diff
@@ -0,0 +1,2075 @@
+cmake_minimum_required(VERSION 3.9.0)
+project(mixxx VERSION 2.3.0)
+set(CMAKE_PROJECT_HOMEPAGE_URL "https://www.mixxx.org")
+set(CMAKE_PROJECT_DESCRIPTION "Mixxx is Free DJ software that gives you everything you need to perform live mixes.")
+
+# Support new IN_LIST if() operator
+if(POLICY CMP0057)
+  cmake_policy(SET CMP0057 NEW)
+endif()
+
+# Let AUTOMOC and AUTOUIC process GENERATED files
+if(POLICY CMP0071)
+  cmake_policy(SET CMP0071 NEW)
+endif()
+
+set(CMAKE_CXX_STANDARD 14)
+include(CMakeDependentOption)
+include(CheckSymbolExists)
+include(ExternalProject)
+include(GNUInstallDirs)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
+
+if(WIN32)
+  # Add support for lib prefix on Windows
+  set(CMAKE_FIND_LIBRARY_PREFIXES "" "lib")
+endif()
+set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+
+# Get the current working branch
+execute_process(
+  COMMAND git rev-parse --abbrev-ref HEAD
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  OUTPUT_VARIABLE GIT_BRANCH
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  ERROR_QUIET
+)
+if(NOT GIT_BRANCH)
+  set(GIT_BRANCH "unknown")
+endif()
+message(STATUS "Git branch: ${GIT_BRANCH}")
+
+# Get the latest abbreviated commit hash of the working branch
+execute_process(
+  COMMAND git log -1 --format=%h
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  OUTPUT_VARIABLE GIT_COMMIT_HASH
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  ERROR_QUIET
+)
+if(NOT GIT_COMMIT_HASH)
+    set(GIT_COMMIT_HASH "unknown")
+endif()
+message(STATUS "Git commit: ${GIT_COMMIT_HASH}")
+
+# Get the number of commits on the working branch
+execute_process(
+  COMMAND git rev-list --count --first-parent HEAD
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  OUTPUT_VARIABLE GIT_COMMIT_COUNT
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  ERROR_QUIET
+)
+if(NOT GIT_COMMIT_COUNT)
+    set(GIT_COMMIT_COUNT "unknown")
+endif()
+
+# Check if the worktree is dirty
+execute_process(
+  COMMAND git diff --quiet
+  RESULT_VARIABLE GIT_WORKTREE_DIRTY
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+  ERROR_QUIET
+)
+if(GIT_WORKTREE_DIRTY EQUAL "1")
+  set(GIT_COMMIT_COUNT "${GIT_COMMIT_COUNT}+")
+endif()
+message(STATUS "Git commit count: ${GIT_COMMIT_COUNT}")
+
+# ccache support
+find_program(CCACHE_FOUND "ccache")
+cmake_dependent_option(CCACHE_SUPPORT "Enable ccache support" ON "CCACHE_FOUND;NOT MSVC" OFF)
+if(CCACHE_SUPPORT)
+  if(MSVC)
+    message(FATAL_ERROR "ccache support is incompatible with MSVC.")
+  elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" # GNU is GNU GCC
+    OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    # without this compiler messages in `make` backend would be uncolored
+    set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -fdiagnostics-color=auto")
+  endif()
+  set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "ccache")
+  set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK "ccache")
+endif()
```

Comment:
Here a else branch could be handy that suggests to install "ccache" to speed up the work with different git branches.

Reply:
IMHO I'd rather add this to the compilation instructions/README instead of adding this message to the build log every time even if it's just a user who likes to compile stuff instead of a developer.

Reply:
I actually stumbled over it, because it is just nothing printed at all. I was not sure if ccache part was even considered. 
This compares not well to other messages that are even redundant.

Reply:
OK, I restructured the code and added some messages.

Reply:
BTW to see available options you can use `cmake - LH <sourcedir>`.